File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> ump;
        int sum = 0;
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            int rem = target - nums[i];
            if(ump.find(rem)!=ump.end()){
                return {ump[rem],i};
            }
            ump[nums[i]] = i;
        }
        return {};
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        map = {}
        for i in nums:
            if i not in map:
                map[i] = 1
            else:
                map[i] = map.get(i)+1
        
        for key in map:
            if map[key]>1:
                return True
        
        return False
    